File: module_java/src/main/java/org/gmfbilu/superapp/module_java/basic/singleton/Singleton7.java
package org.gmfbilu.superapp.module_java.basic.singleton;

/**
 * 双重校验锁,避免在多线程环境中被创建多次
 */
public class Singleton7 {
    private volatile static Singleton7 singleton;

    private Singleton7() {
    }

    public static Singleton7 getSingleton7() {
        if (singleton == null) {
            synchronized (Singleton7.class) {
                if (singleton == null) {
                    singleton = new Singleton7();
                }
            }
        }
        return singleton;
    }
}

File: module_kotlin/src/main/java/org/gmfbilu/superapp/module_kotlin/kotlin/helloWorld/设计模式/Singleton7.kt
package org.gmfbilu.superapp.module_kotlin.kotlin.helloWorld.设计模式

/**
 * 双重校验锁,避免在多线程环境中被创建多次
 */
class Singleton7 private constructor() {

    companion object {
        val instance by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) {
            Singleton7()
        }
    }
}